cpp/src/ast/transform.cu

```diff
@@ -88,45 +88,54 @@ std::unique_ptr<column> compute_column(table_view const table,
                                        rmm::cuda_stream_view stream,
                                        rmm::mr::device_memory_resource* mr)
 {
-  auto const expr_linearizer = linearizer(expr, table);                // Linearize the AST
-  auto const plan            = ast_plan{expr_linearizer, stream, mr};  // Create ast_plan
+  // Prepare output column. Whether or not the output column is nullable is
+  // determined by whether any of the columns in the input table are nullable.
+  // If none of the input columns actually contain nulls, we can still use the
+  // non-nullable version of the expression evaluation code path for
+  // performance, so we capture that information as well.
+  auto nullable =
+    std::any_of(table.begin(), table.end(), [](column_view c) { return c.nullable(); });
+  auto has_nulls = nullable && std::any_of(table.begin(), table.end(), [](column_view c) {
```

Comment:
This doesn't work for a couple of reasons, one cosmetic and one substantial. The cosmetic issue is that `table_view` doesn't define `const` iterators. That problem could be amended (although out of scope for this PR), but the deeper problem is that `column_view.has_nulls` is not actually a `const` function because it triggers a null count if needed.

Reply:
Okay, I see. You might be able to adapt some bits from my suggestion. The method `column_view.nullable()` [is const](https://github.com/rapidsai/cudf/blob/0206fc9ad1c394dd694a3ddd8d9460f3c2442be0/cpp/include/cudf/column/column_view.hpp#L147), so you could still use `[](const column_view& c){ ... }` for the first check for nullability. The result booleans can also be `const`.

Reply:
Yep I made the output variables const, but didn't make the parameters const. I could modify the lambda parameters in the first case as well. I don't think it makes a huge difference either way in that case.